doc/sphinxext/allow_nan.py

```diff
@@ -0,0 +1,68 @@
+from sklearn.utils import all_estimators
+from sklearn.compose import ColumnTransformer
+from sklearn.pipeline import FeatureUnion
+from sklearn.decomposition import SparseCoder
+from sklearn.utils.estimator_checks import _construct_instance
+from docutils import nodes
+import warnings
+
+from docutils.parsers.rst import Directive
+
+
+class Allow_Nan(Directive):
+    @staticmethod
+    def make_paragraph_for_estimator_type(estimator_type):
+        intro = nodes.list_item()
+        intro += nodes.strong(text="Estimators that allow NaN values for type ")
+        intro += nodes.literal(text=f"{estimator_type}")
+        intro += nodes.strong(text=":\n")
+        exists = False
+        lst = nodes.bullet_list()
+        for name, est_class in all_estimators(type_filter=estimator_type):
+            try:
+                est = _construct_instance(est_class)
+            except:
```

Comment:
I think this raises a `SkipTest` exception. In those cases, I think we can skip without warning, since warning is not that actionable here.

```suggestion
from contextlib import supress
...
            with suppress(SkipTest):
                est = _construct_instance(est_class)
```

I suspect the "Estimators that handle NaN values" section is for non meta-estimators anyways.

Reply:
Make sense! I've updated code as suggested :)